Docker doesn't build on aarch64

I am using my RPi4

```bash
$ sudo docker build -t dimension:latest .
Sending build context to Docker daemon  7.463MB
Step 1/22 : FROM node:12.16.1-alpine AS builder
 ---> be7c0ffeb20b
Step 2/22 : LABEL maintainer="Andreas Peters <support@aventer.biz>"
 ---> Using cache
 ---> cef2b6bf3323
Step 3/22 : WORKDIR /home/node/matrix-dimension
 ---> Using cache
 ---> 7a47353b8c6c
Step 4/22 : RUN mkdir -p /home/node/matrix-dimension
 ---> Using cache
 ---> 2ab1e30f384f
Step 5/22 : COPY . /home/node/matrix-dimension
 ---> 6bf862a994f6
Step 6/22 : RUN chown -R node /home/node/matrix-dimension
 ---> Running in e23fdbadf8ed
Removing intermediate container e23fdbadf8ed
 ---> a16f704970d1
Step 7/22 : USER node
 ---> Running in 4bf1bee67d13
Removing intermediate container 4bf1bee67d13
 ---> 09db33e6db9b
Step 8/22 : RUN npm clean-install &&     node /home/node/matrix-dimension/scripts/convert-newlines.js /home/node/matrix-dimension/docker-entrypoint.sh  &&     NODE_ENV=production npm run-script build
 ---> Running in 66b61c89bb39

> core-js@2.6.9 postinstall /home/node/matrix-dimension/node_modules/babel-runtime/node_modules/core-js
> node scripts/postinstall || echo "ignore"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon: 
> https://opencollective.com/core-js 
> https://www.patreon.com/zloirock 

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)


> sqlite3@4.2.0 install /home/node/matrix-dimension/node_modules/sqlite3
> node-pre-gyp install --fallback-to-build


node-pre-gyp WARN Using request for node-pre-gyp https download 
node-pre-gyp WARN Tried to download(403): https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v4.2.0/node-v72-linux-arm64.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for sqlite3@4.2.0 and node@12.16.1 (node-v72 ABI, musl) (falling back to source compile with node-gyp) 
gyp ERR! find Python 
gyp ERR! find Python Python is not set from command line or npm configuration
gyp ERR! find Python Python is not set from environment variable PYTHON
gyp ERR! find Python checking if "python" can be used
gyp ERR! find Python - "python" is not in PATH or produced an error
gyp ERR! find Python checking if "python2" can be used
gyp ERR! find Python - "python2" is not in PATH or produced an error
gyp ERR! find Python checking if "python3" can be used
gyp ERR! find Python - "python3" is not in PATH or produced an error
gyp ERR! find Python 
gyp ERR! find Python **********************************************************
gyp ERR! find Python You need to install the latest version of Python.
gyp ERR! find Python Node-gyp should be able to find and use Python. If not,
gyp ERR! find Python you can try one of the following options:
gyp ERR! find Python - Use the switch --python="/path/to/pythonexecutable"
gyp ERR! find Python   (accepted by both node-gyp and npm)
gyp ERR! find Python - Set the environment variable PYTHON
gyp ERR! find Python - Set the npm configuration variable python:
gyp ERR! find Python   npm config set python "/path/to/pythonexecutable"
gyp ERR! find Python For more information consult the documentation at:
gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
gyp ERR! find Python **********************************************************
gyp ERR! find Python 
gyp ERR! configure error 
gyp ERR! stack Error: Could not find any Python installation to use
gyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:307:47)
gyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:136:21)
gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:179:16)
gyp ERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:271:16)
gyp ERR! stack     at exithandler (child_process.js:310:5)
gyp ERR! stack     at ChildProcess.errorhandler (child_process.js:322:5)
gyp ERR! stack     at ChildProcess.emit (events.js:311:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:273:12)
gyp ERR! stack     at onErrorNT (internal/child_process.js:469:16)
gyp ERR! stack     at processTicksAndRejections (internal/process/task_queues.js:84:21)
gyp ERR! System Linux 5.4.0-1026-raspi
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--module=/home/node/matrix-dimension/node_modules/sqlite3/lib/binding/node-v72-linux-arm64/node_sqlite3.node" "--module_name=node_sqlite3" "--module_path=/home/node/matrix-dimension/node_modules/sqlite3/lib/binding/node-v72-linux-arm64" "--napi_version=5" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v72"
gyp ERR! cwd /home/node/matrix-dimension/node_modules/sqlite3
gyp ERR! node -v v12.16.1
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/home/node/matrix-dimension/node_modules/sqlite3/lib/binding/node-v72-linux-arm64/node_sqlite3.node --module_name=node_sqlite3 --module_path=/home/node/matrix-dimension/node_modules/sqlite3/lib/binding/node-v72-linux-arm64 --napi_version=5 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/home/node/matrix-dimension/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:311:20)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1021:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
node-pre-gyp ERR! System Linux 5.4.0-1026-raspi
node-pre-gyp ERR! command "/usr/local/bin/node" "/home/node/matrix-dimension/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /home/node/matrix-dimension/node_modules/sqlite3
node-pre-gyp ERR! node -v v12.16.1
node-pre-gyp ERR! node-pre-gyp -v v0.11.0
node-pre-gyp ERR! not ok 
Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/home/node/matrix-dimension/node_modules/sqlite3/lib/binding/node-v72-linux-arm64/node_sqlite3.node --module_name=node_sqlite3 --module_path=/home/node/matrix-dimension/node_modules/sqlite3/lib/binding/node-v72-linux-arm64 --napi_version=5 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72' (1)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sqlite3@4.2.0 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sqlite3@4.2.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2021-01-12T14_46_13_904Z-debug.log
The command '/bin/sh -c npm clean-install &&     node /home/node/matrix-dimension/scripts/convert-newlines.js /home/node/matrix-dimension/docker-entrypoint.sh  &&     NODE_ENV=production npm run-script build' returned a non-zero code: 1
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,9 +7,7 @@ WORKDIR /home/node/matrix-dimension
 
 RUN mkdir -p /home/node/matrix-dimension
 
-RUN apk update && \
-    apk add python2 glib-dev make g++ vips-dev libc-dev && \
-    rm -rf /var/lib/apk/* /var/cache/apk/*
+RUN apk --no-cache add python2 glib-dev make g++ vips-dev libc-dev
 
 COPY . /home/node/matrix-dimension
 
@@ -37,9 +35,7 @@ RUN chown -R node /home/node/matrix-dimension
 
 RUN mkdir /data && chown -R node /data
 
-RUN apk update && \
-    apk add python2 glib-dev make g++ vips-dev libc-dev && \
-    rm -rf /var/lib/apk/* /var/cache/apk/*
+RUN apk --no-cache add python2 glib-dev make g++ vips-dev libc-dev
 
 ENV CPATH=/usr/include/glib-2.0:/usr/lib/glib-2.0/include/
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,11 +7,17 @@ WORKDIR /home/node/matrix-dimension
 
 RUN mkdir -p /home/node/matrix-dimension
 
+RUN apk update && \
+    apk add python2 glib-dev make g++ vips-dev libc-dev && \
+    rm -rf /var/lib/apk/* /var/cache/apk/*
+
 COPY . /home/node/matrix-dimension
 
 RUN chown -R node /home/node/matrix-dimension
 
 USER node
+
+ENV CPATH=/usr/include/glib-2.0:/usr/lib/glib-2.0/include/
 
 RUN npm clean-install && \
     node /home/node/matrix-dimension/scripts/convert-newlines.js /home/node/matrix-dimension/docker-entrypoint.sh  && \
@@ -30,6 +36,12 @@ COPY --from=builder /home/node/matrix-dimension/config /home/node/matrix-dimensi
 RUN chown -R node /home/node/matrix-dimension
 
 RUN mkdir /data && chown -R node /data
+
+RUN apk update && \
+    apk add python2 glib-dev make g++ vips-dev libc-dev && \
+    rm -rf /var/lib/apk/* /var/cache/apk/*
+
+ENV CPATH=/usr/include/glib-2.0:/usr/lib/glib-2.0/include/
 
 USER node
 

